<?xml version="1.0" encoding="UTF-8" ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org./TR/xhtml1/DTD/hxtml1-strict.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">

<head>
<title>Imputation-based HLA typing with SNPs in GWAS studies</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="description" content="Imputation-based HLA typing with SNPs in GWAS studies">
<meta name="keywords" content="HLA, MHC, Imputation, SNP, GWAS, HIBAG">
</head>


<body>
<h1>Imputation-based HLA typing with SNPs in GWAS studies</h1>

<p><font size="4">
<br>Email: <a href="mailto:zhengx@u.washington.edu">Dr. Xiuwen Zheng</a>
<br>This page was last updated on <b>Aug 30, 2017</b>
</font></p>

<h2 id="contents">Contents</h2>
<font size="4"><ul>
	<li><a href="#intro">Introduction</a></li>
	<li><a href="#download">Download</a></li>
	<li><a href="#citation">Citation</a></li>
	<li><a href="#link">Resources</a></li>
</ul></font>
<p>&nbsp;</p>
<hr>
<span style="FONT-FAMILY: 'Times New Roman','serif'; FONT-SIZE: 13.5pt">


<h2 id="intro">Introduction</h2>
<p><font size="4">
SNP-based imputation approaches for human leukocyte antigen (HLA) typing take advantage of the extended haplotype structure within the MHC to predict classical HLA alleles using dense SNP genotypes, such as those available from SNP panels of genome- wide association study (GWAS). These methods enable HLA analyses on existing SNP datasets genotyped in GWAS studies at no extra cost. Here, I describe the workflow of HIBAG for obtaining a sample’s HLA class I and II genotypes of two-field resolution using SNP data. Two examples are provided to illustrate with a publicly available HLA and SNP dataset: genotype imputation with pre-fit classifiers in GWAS, and model training to build a new classifier.
</font></p><a href="#contents">back to contents</a><p>&nbsp;</p>
<hr>


<h2 id="download">Download</h2>
<p><font size="4">
HIBAG package: <a href="http://www.bioconductor.org/packages/HIBAG">http://www.bioconductor.org/packages/HIBAG</a>
<br>
HIBAG.gpu package: <a href="https://github.com/zhengxwen/HIBAG.gpu">https://github.com/zhengxwen/HIBAG.gpu</a>
<br>
ImmPuteDataPackage.zip: <a href="./ImmPuteDataPackage.zip">download</a> (or download from <a href="http://immpute-project.immunogenomics.org">http://immpute-project.immunogenomics.org</a>)
<br>
ImmunoChip-Broad-HLARES-HLA4-hg19.RData: <a href="./ImmunoChip-Broad-HLARES-HLA4-hg19.RData">download</a> (or download from <a href="http://zhengxwen.github.io/HIBAG/platforms.html">http://zhengxwen.github.io/HIBAG/platforms.html</a>)
</font>
<p><a href="#contents">back to contents</a></p><p>&nbsp;</p>
<hr>


<h2 id="citation">Citation</h2>
	If you use HIBAG in a published analysis, please report the HIBAG version used and cite
the appropriate publication or publications listed below:
<ul>
<li><div>
	Zheng X, Shen J, Cox C, Wakefield J, Ehm M, Nelson M, Weir BS.
	HIBAG – HLA Genotype Imputation with Attribute Bagging.
	Pharmacogenomics Journal (2014).
	<a href="http://dx.doi.org/10.1038/tpj.2013.18">doi: 10.1038/tpj.2013.18.</a>
</div></li>
</ul> 
<p><a href="#contents">back to contents</a></p><p>&nbsp;</p>
<hr>


<h2 id="link">Resources</h2>
<ul>
	<li>Downloadable pre-fit classifiers: <a href="http://zhengxwen.github.io/HIBAG/hibag_index.html">http://zhengxwen.github.io/HIBAG/hibag_index.html</a>
	<li>Allele Frequency Net Database (AFND): <a href="http://www.allelefrequencies.net">http://www.allelefrequencies.net</a>
	<li>IMGT/HLA Database: <a href="http://www.ebi.ac.uk/imgt/hla">http://www.ebi.ac.uk/imgt/hla</a>
	<li>HLA Nomenclature: <a href="http://hla.alleles.org/alleles/g_groups.html">G Codes</a> and <a href="http://hla.alleles.org/alleles/p_groups.html">P Codes</a> for reporting of ambiguous allele typings
	<li>GlaxoSmithKline (GSK): <a href="http://www.gsk.com">http://www.gsk.com</a>
	<li>International HLA and Immunogenetics Workshop (IHIWS): <a href="http://ihiws.org">http://ihiws.org</a>
</ul>
<p><a href="#contents">back to contents</a></p><p>&nbsp;</p>
<hr>


<!-- Start of StatCounter Code for Default Guide -->
<script type="text/javascript">
	var sc_project=8185875; 
	var sc_invisible=0; 
	var sc_security="c902b96b"; 
</script>
<script type="text/javascript" src="http://www.statcounter.com/counter/counter.js"></script>
<noscript><div class="statcounter">
<a title="visit tracker on tumblr" href="http://statcounter.com/tumblr/" target="_blank">
	<img class="statcounter" src="http://c.statcounter.com/8185875/0/c902b96b/0/" alt="visit tracker on tumblr">
</a></div></noscript>
<!-- End of StatCounter Code for Default Guide -->

</body>
</html>
